test-data/unit/check-based-typeguard.test

```diff
@@ -23,3 +23,24 @@ a: object
 assert (lambda: f(a))()
 reveal_type(a)  # N: Revealed type is "int"
 [builtins fixtures/tuple.pyi]
+
+
+[case testSubtype]
+from typing_extensions import TypeGuard
+from typing import Callable
+def takes_typeguard(x: Callable[[object], TypeGuard[int]]): ...
+def typeguard1(x: object) -> 'x' is int: ...
+def typeguard2(x: object, y: object) -> 'y' is int: ...
+
+takes_typeguard(typeguard1)
+takes_typeguard(typeguard2)  # E: Argument 1 to "takes_typeguard" has incompatible type "(x: object, y: object) -> y is int"; expected "(object) -> first argument is int"  [arg-type]
+[builtins fixtures/tuple.pyi]
+
+
+[case testTypeguardSubtype]
+def typeguard(x: object) -> 'x' is int: ...
+
+a: object
+assert (lambda y: typeguard(y))(a)
+reveal_type(a)  # N: Revealed type is "int"
+[builtins fixtures/tuple.pyi]
```

Comment:
add tests for:

- arguments other than the first one
- keyword args
- non-overlapping types
- referring to variable that doesn't exist (eg. `def foo(a: object) -> b is int`)
- referring to type that doesn't exist?

Reply:
hows that?